dll/win32/browseui/explorerband.cpp

```diff
@@ -591,46 +524,176 @@ LRESULT CExplorerBand::OnShellEvent(UINT uMsg, WPARAM wParam, LPARAM lParam, BOO
     return 0;
 }
 
+BOOL
+CExplorerBand::IsTreeItemInEnum(
+    HTREEITEM hItem,
+    IEnumIDList *pEnum)
+{
+    NodeInfo* pNodeInfo = GetNodeInfo(hItem);
+    if (!pNodeInfo)
+        return FALSE;
+
+    pEnum->Reset();
+
+    CComHeapPtr<ITEMIDLIST_RELATIVE> pidlTemp;
+    while (pEnum->Next(1, &pidlTemp, NULL) == S_OK)
+    {
+        if (ILIsEqual(pidlTemp, pNodeInfo->relativePidl))
+            return TRUE;
```

Comment:
If the function returns true, what happens to pidlTemp?

Reply:
Freed by `CComHeapPtr<ITEMIDLIST_RELATIVE>` dtor.